neqo-http3/src/headers_checks.rs

```diff
@@ -48,12 +48,12 @@ impl TryFrom<(MessageType, &str)> for PseudoHeaderState {
 /// # Errors
 /// Returns an error if response headers do not contain
 /// a status header or if the value of the header cannot be parsed.
-pub fn is_interim(headers: &[Header]) -> Res<bool> {
+pub fn is_interim(headers: &[Header]) -> Res<(bool, i32)> {
     let status = headers.iter().take(1).find(|h| h.name() == ":status");
     if let Some(h) = status {
         #[allow(clippy::map_err_ignore)]
         let status_code = h.value().parse::<i32>().map_err(|_| Error::InvalidHeader)?;
-        Ok((100..200).contains(&status_code))
+        Ok(((100..200).contains(&status_code), status_code))
```

Comment:
Would it be possible to do this instead?

```suggestion
        if status_code == 101 {
            Err(Error::InvalidHeader)
        } else {
            Ok((100..200).contains(&status_code))
        }
```

That would integrate better into existing code.  It means generating an error when sending a message, but I think that's good too.

Reply:
Right, doing this means less change for the current code.  Thanks.